internal/resolution/variablesources/required_package.go

```diff
@@ -91,15 +91,11 @@ func (r *RequiredPackageVariableSource) GetVariables(ctx context.Context) ([]dep
 }
 
 func (r *RequiredPackageVariableSource) notFoundError() error {
-	// TODO: update this error message when/if we decide to support version ranges as opposed to fixing the version
-	//  context: we originally wanted to support version ranges and take the highest version that satisfies the range
-	//  during the upstream call on the 2023-04-11 we decided to pin the version instead. But, we'll keep version range
-	//  support under the covers in case we decide to pivot back.
 	if r.versionRange != "" && r.channelName != "" {
-		return fmt.Errorf("package '%s' at version '%s' in channel '%s' not found", r.packageName, r.versionRange, r.channelName)
+		return fmt.Errorf("package '%s' matching version '%s' in channel '%s' not found", r.packageName, r.versionRange, r.channelName)
```

Comment:
nit: A more idiomatic English clause might be 

> no package matching ... found

Reply:
@stevekuznetsov Sounds better. Should we update other versions to match the same style?

* When there is only package name: `no package '%s' found`
* When there is package name and version range: `no package '%s' matching version '%s' found`
* When there is package name and channel: `no package '%s' in channel '%s' found`
* When there is package name, channel and version range: `no package '%s' matching version '%s' in channel '%s' found`

Or perhaps something like this for the last two involving channel:
* `no package '%s' found in channel '%s'`
* `no package '%s' matching version '%s' found in channel '%s'`

Reply:
I think those sound better, yes - your suggestion with the channel clause at the end is also good!

Reply:
Updated, please take another look